src/Raven.Server/Logging/RavenLogManagerServerExtensions.cs

```diff
@@ -0,0 +1,796 @@
+        {
+            if (configuration.Logs.ConfigPath == null)
+                return false;
+
+#if !RVN
+            AsyncHelpers.RunSync(() => InstallAdditionalTargetsAsync(configuration));
+#endif
+            LogManager.Setup(x => x.LoadConfigurationFromFile(configuration.Logs.ConfigPath.FullPath, optional: false));
+            var c = LogManager.Configuration;
+#if !RVN
+            c.AddRule(AdminLogsRule);
+#endif
+            DefaultRule = c.FindRuleByName(Constants.Logging.Names.DefaultRuleName);
+            if (DefaultRule == null)
+                ThrowNoRule(Constants.Logging.Names.DefaultRuleName);
+
+            DefaultAuditRule = c.FindRuleByName(Constants.Logging.Names.DefaultAuditRuleName);
+            if (DefaultAuditRule == null)
+                ThrowNoRule(Constants.Logging.Names.DefaultAuditRuleName);
+
+            SystemRule = c.FindRuleByName(Constants.Logging.Names.SystemRuleName);
+            if (SystemRule == null)
+                ThrowNoRule(Constants.Logging.Names.SystemRuleName);
+
+            MicrosoftRule = c.FindRuleByName(Constants.Logging.Names.MicrosoftRuleName);
+            if (MicrosoftRule == null)
+                ThrowNoRule(Constants.Logging.Names.MicrosoftRuleName);
+
+            LogManager.Setup(x => x.LoadConfiguration(c));
+            LogManager.ReconfigExistingLoggers(purgeObsoleteLoggers: true);
+
+            if (Logger.IsInfoEnabled)
+                Logger.Info($"Logging configured from '{configuration.Logs.ConfigPath}' configuration file and set to [{DefaultRule.Levels.FirstOrDefault() ?? LogLevel.Off}, {DefaultRule.Levels.LastOrDefault() ?? LogLevel.Off}] level.");
+
+            return true;
+        }
+
+        void ConfigureInternalLogging()
+        {
+            if (configuration.Logs.NLogInternalPath == null)
+                return;
+
+            InternalLogger.LogFile = configuration.Logs.NLogInternalPath.Combine("nlog.internal.log").FullPath;
+            InternalLogger.LogLevel = configuration.Logs.NLogInternalLevel.ToNLogLogLevel();
+            InternalLogger.LogToConsole = configuration.Logs.NLogLogToConsole;
+            InternalLogger.LogToConsoleError = configuration.Logs.NLogLogToConsoleError;
+        }
+
+        static void ThrowNoRule(string ruleName)
+        {
+            throw new InvalidOperationException($"Could not find '{ruleName}' rule in the configuration file.");
+        }
+    }
+
+#if !RVN
+    public static void ConfigureAuditLog(this RavenLogManager logManager, RavenServer server, RavenLogger logger)
+    {
+        var configuration = server.Configuration;
+
+        if (configuration.Security.AuditLogPath == null)
+            return;
+
+        if (configuration.Security.AuthenticationEnabled == false)
+        {
+            if (logger.IsErrorEnabled)
+                logger.Error("The audit log configuration 'Security.AuditLog.FolderPath' was specified, but the server is not running in a secured mode. Audit log disabled!");
```

Comment:
Please use `RavenConfiguration.GetKey()` instead of hard coding `'Security.AuditLog.FolderPath'`